### DISCUSSION TITLE
has anyone managed to get gnome polkit agent working correctly ?

### DISCUSSION THREAD

**Discussion by Hanibachi** [YaLTeR/niri#906] [Q&A]
debian 12 sid with policykit-1-gnome installed, works fine on gnome but not in niri, i know gnpme auth agent isnt known for being the most compatible but i'm curious if anyone got it running ?

**Comment by YaLTeR** [comment]
GNOME does not use it. It has a polkit dialog built into the GNOME Shell. The "GNOME polkit agent" has been deprecated many years ago.

**Comment by nebulosa2007** [comment]
I'm using [polkit-gnome-authentication-agent-1](https://github.com/nebulosa2007/archinstallbtfrs/blob/master/dotfiles/niri_config/systemd/user/gnome-polkit-agent.service) which runs on login to Niri. Then when `waybar` runs script on a click and there is a `sudo` command in the script, polkit shows a password popup.

Yes, it's outdated, but the package is still being updated by the maintainers.

**Comment by approximate** [comment]
I tried and failed to make gnome- or other polkit agents work. Eventually installed [soteria](https://github.com/imvaskel/soteria), which worked immediately after running `niri msg action spawn -- soteria`.

**Reply by norveganon** [reply]
One thing is to have it working, but is it declared safe and secure by someone who knows what they're talking about? I don't, so I would feel better about switching to it with a 3rd-party deeming it secure, you know?

**Reply by approximate** [reply]
sure, i can understand the concern. 

on the other hand though, the source code is there. even if you can't do a proper security audit and safety assession unless you're an expert, you can at least make sure the code doesn't do anything obviously malicious (e.g. phoning home) or wrong (e.g. writing passwords to plain text files). for my usecase this is an acceptable level of risk. 

i much rather spend 5 minutes setting this up and go on with my day than waste a week trying to find and configure another polkit agent that is half-recent, doesn't have a metric ton of dependencies, and actually works.

**Reply by norveganon** [reply]
Made the switch, works great. I couldn't do `niri msg action spawn -- soteria` though, but rather `niri msg action spawn -- /usr/lib/soteria-polkit/soteria`.

**Comment by noahpolimon** [comment]
switched to niri yesterday and kept some hyprland things that i was using. hyprpolkitagent (and most of the other hypr stuff that dont depend directly on hyprland) works so far.
